### PR TITLE
docs: add rego templates

### PR DESCRIPTION
**Pull request by binbin-li**
1. Add 2 templates that verify signatures signed by multiple trusted identities.
2. Move original templates to the new rego-templates.md
3. Add a note to the last example as verifier name is not correctly set up.

Fix: https://github.com/deislabs/ratify/issues/1049

### CODE REVIEW

File: docs/reference/rego-templates.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+### Example 4
+I trust multiple identities, and need to find at least one valid signature signed by each identity.
+
+Prerequisites:
+
+1. The artifact is only signed by Notation, there might be a few signatures attached to the artifact.
+2. Users has configured different Notation verfiers, and each verifier has a different identity.
+
+Note: The below Rego query currently does not work as expected as it's blocked by a known [bug](https://github.com/deislabs/ratify/issues/1066) which reports the same name for different Notation verifiers.
```
Comment (akashsinghal) [inline docs/reference/rego-templates.md:178]: Do you think we should wait to add this example until after 1066 has been addressed?

File: docs/reference/rego-templates.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+# should be at least one verifier report that is successful
+failed_verify(reports) if {
+	[path, value] := walk(reports)
+	path[count(path) - 1] == "verifierReports"
+	count(value) == 0
+}
+```
+
+### Example 3
+I trust multiple identities, and need to find at least one valid signature signed by any identity.
```
Comment (susanshi) [inline docs/reference/rego-templates.md:133]: Q:  is example3 and 4 the same ? "I trust multiple identities, and need to find at least one valid signature signed by any identity."

File: docs/reference/rego-templates.md (hunk 1 above)
Comment (binbin-li) [reply docs/reference/rego-templates.md:178]: I agree with you waiting until 1066 gets fixed. @FeynmanZhou, as we discussed in previous meetings, we'd like to have these examples in v1. Do you think we can wait for next release?

File: docs/reference/rego-templates.md (hunk 2 above)
Comment (binbin-li) [reply docs/reference/rego-templates.md:133]: No, they're different. example 3 is `by any identity` while example 4 is `by each identity`. In example 3, only one signature is required, signed by any of the available identities. Conversely, in example 4, signatures must be signed by all trusted identities.
Do you have any suggestion to rephrase it?

File: docs/reference/rego-templates.md (hunk 1 above)
Comment (binbin-li) [reply docs/reference/rego-templates.md]: btw, I have removed this example, I'll add it back once we fix the blocking issue.

**Review by akashsinghal — APPROVED** [review]

**Pull request merged**